functions/ConvertTo-DbaTimeline.ps1

```diff
@@ -0,0 +1,191 @@
+            Write-Output "Ummm, we do not know how to create chart for this. Sorry.`r`nDo you want to help us understand what is that you are trying to do? Get in touch!"; break;
+        }      
+        #build html container
+@"
+<html>
+<head>
+<!-- Developed by Marcin Gminski, https://marcin.gminski.net, 2018 -->
+<!-- Load jQuery required to autosize timeline -->
+<script src="https://code.jquery.com/jquery-3.3.1.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
+<!-- Load Bootstrap -->
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">
+<!-- Load Google Charts library -->
+<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+<!-- a bit of custom styling to work with bootstrap grid -->
+<style>
+
+    html,body{height:100%;background-color:#c2c2c2;}
+    .viewport {height:100%}
+
+    .chart{
+        background-color:#fff;
+        text-align:left;
+        padding:0;
+        border:1px solid #7D7D7D;
+        -webkit-box-shadow:1px 1px 3px 0 rgba(0,0,0,.45);
+        -moz-box-shadow:1px 1px 3px 0 rgba(0,0,0,.45);
+        box-shadow:1px 1px 3px 0 rgba(0,0,0,.45)
+    }
+    .badge-custom{background-color:#939}
+    .container {
+        height:100%;
+    }
+    .fill{
+        width:100%;
+        height:100%;
+        min-height:100%;
+        padding:10px;
+    }
+    .timeline-tooltip{
+        border:1px solid #E0E0E0;
+        font-family:Arial,Helvetica;
+        font-size:10pt;
+        padding:12px
+    }
+    .timeline-tooltip div{padding:6px}
+    .timeline-tooltip span{font-weight:700}
+</style>
+    <script type="text/javascript">
+    google.charts.load('43', {'packages':['timeline']});
+    google.charts.setOnLoadCallback(drawChart);
+    function drawChart() {
+        var container = document.getElementById('Chart');
+        var chart = new google.visualization.Timeline(container);
+        var dataTable = new google.visualization.DataTable();
+        dataTable.addColumn({type: 'string', id: 'vLabel'});
+        dataTable.addColumn({type: 'string', id: 'hLabel'});
+        dataTable.addColumn({type: 'string', role: 'style' });
+        dataTable.addColumn({type: 'date', id: 'date_start'});
+        dataTable.addColumn({type: 'date', id: 'date_end'});
+
+        dataTable.addRows([
+"@
+    }
+
+    process
```

Comment:
because piped types can't be detected in `begin`, I'll build up the output then output it all at the end.